ecpy_hqc_legacy/instruments/starters/legacy_starter.py

```diff
@@ -73,9 +73,16 @@ def format_connection_infos(self, infos):
         """Use pyvisa to build the canonical resource name.
 
         """
+#        from pyvisa.rname import assemble_canonical_name
+#        infos = {k: v for k, v in infos.items() if v}
+#        return {'resource_name': assemble_canonical_name(**infos)}
         from pyvisa.rname import assemble_canonical_name
         infos = {k: v for k, v in infos.items() if v}
-        return {'resource_name': assemble_canonical_name(**infos)}
+        try:
+            return {'resource_name': infos['resource_name']}
+        except:
```

Comment:
Never use a bare except !
Bare except catches SystemExit and KeyboardInterrupt which you really does not want to catch.
Furthermore here a simple check for the presence of 'resource_name' in the infos would be enough to determine whether or not to assemble the name. I would prefer:
```python
if 'resource_name' in infos:
    return {'resource_name': infos['resource_name']}
else:
    return return {'resource_name': assemble_canonical_name(**infos)}
```